series.c

```diff
@@ -509,14 +508,17 @@ static char *readconfigfile(fileinfo *file, gameseries *series)
  * list stored under the second argument. This function is used as a
  * findfiles() callback.
  */
-static int getseriesfile(char const *filename, void *data)
+static int getseriesfile(char const *ogfilename, void *data)
 {
     fileinfo		file;
     seriesdata	       *sdata = (seriesdata*)data;
     gameseries	       *series;
     unsigned long	magic;
     char	       *datfilename;
     int			config, f;
+	char        *filename = malloc(strlen(ogfilename) - strlen(sdata->curdir));
+
+	strcpy(filename, ogfilename + strlen(sdata->curdir) + 1);
```

Comment:
`filename` is never freed, but it doesn't need to be allocated/copied to begin with.  Instead, you can just take a pointer to the part of the filename past the last slash.